Modules/Loadable/Markups/qSlicerMarkupsModule.cxx

```diff
@@ -119,6 +133,24 @@ void qSlicerMarkupsModule::setup()
 {
   this->Superclass::setup();
 
+   vtkMRMLMarkupsRegistrationFactory::GetInstance()->RegisterMarkup(
+     vtkMRMLMarkupsAngleNode::New(), vtkSlicerAngleWidget::New());
```

Comment:
Ideally, we should only register the widget and the method would be renamed from `RegisterMarkup` to `RegisterMarkupsWidget`

Similarly, the class `vtkMRMLMarkupsRegistrationFactory` would be renamed `vtkMRMLMarkupsWidgetFactory`

Reply:
Thanks @jcfr . In our last meeting we decided to try to integrate the registration in the markups logic itself, but I will keep this idea of only registering the widget when I propose the new changes.

Reply:
As I see it at this point, we need the node-widget association. The displayable manager will try to decide which widget to create, based on the display node. However, several markups are sharing a common display node. For disambiguation one can retrieve the `vtkMRMLMarkupsNode` from the `vtkMRMLDisplayNode` and check with the registered `vtkMRMLarkupsNode`s.

Alternatively, instead of keeping the association on the registration, I think we could keep the associated vtkMRMLMarkupsNode on the `vtkSlicerMarkupsWidget`. @jcfr, @lassoan let me know your thoughts on this.